File: application/app/domain/repositories/EndpointRepository.scala
package domain.repositories

import domain.model.Endpoint

import scala.concurrent.Future

trait EndpointRepository {
  def getAll(): Future[Seq[Endpoint]]
  def getById(id: String): Future[Option[Endpoint]]
  def create(endpoint: Endpoint): Future[Endpoint]
  def update(endpoint: Endpoint): Future[Endpoint]
  def delete(id: String): Future[Void]
}

File: ws-gateway-application/src/main/java/com/cosmin/wsgateway/application/repositories/EndpointRepository.java
package com.cosmin.wsgateway.application.repositories;

import com.cosmin.wsgateway.domain.model.Endpoint;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.Optional;

public interface EndpointRepository {
    Flux<Endpoint> getAll();
    Mono<Optional<Endpoint>> getById(String id);
    Mono<Endpoint> save(Endpoint endpoint);
    Mono<Endpoint> deleteById(String id);
}
